views/js/runner/testStore.js

```diff
@@ -0,0 +1,295 @@
+                            }
+                        },
+
+                        /**
+                         * Set an item with the given key
+                         * @param {String} key - the item key
+                         * @param {*} value - the item value
+                         * @returns {Promise<Boolean>} with true in resolve if added/updated
+                         */
+                        setItem : function setItem(key, value){
+                            return loadedStore.setItem(storeKey(key), value);
+                        },
+
+                        /**
+                         * Remove an item with the given key
+                         * @param {String} key - the item key
+                         * @returns {Promise<Boolean>} with true in resolve if removed
+                         */
+
+                        removeItem : function removeItem(key){
+                            return loadedStore.removeItem(storeKey(key));
+                        },
+
+                        /**
+                         * Clear the current store
+                         * @returns {Promise<Boolean>} with true in resolve once cleared
+                         */
+                        clear : function clear(){
+                            if(unifiedStore){
+                                return loadedStore.getItems()
+                                    .then(function(entries){
+                                        _.forEach(entries, function(entry, key){
+                                            if(keyPattern.test(key)){
+                                                loadedStore.removeItem(key);
+                                            }
+                                        });
+                                    });
+                            } else {
+                                return loadedStore.clear();
+                            }
+                        }
+                    };
+                });
+            },
+
+            /**
+             * Define the given store as "volatile".
+             * It means the store data can be revoked
+             * if the user change browser for example
+             * @param {String} storeName - the name of the store to set as volatile
+             * @returns {testStore} chains
+             */
+            setVolatile : function setVolatile(storeName){
+                if(!_.contains(volatiles, storeName)){
+                    volatiles.push(storeName);
+                }
+                return this;
+            },
+
+            /**
+             * Check the given storeId. If different from the current stored identifier
+             * we initiate the invalidation of the volatile data.
+             * @param {String} storeId - the id to check
+             * @returns {Promise<Boolean>} true if cleared
+             */
+            clearVolatileOnStoreChange : function clearVolatileOnStoreChange(storeId){
```

Comment:
Also when reading this method title I was expecting some kind of listener, or that it would be bound to a listener, but it seems that it is just called as a regular function in the provider init. To avoid confusion it may be better named something like `clearVolatileIfStoreChanged`